test/TestPercentageMath.sol

```diff
@@ -64,6 +82,37 @@ contract TestPercentageMath is Test {
         assertEq(PercentageMath.percentDiv(x, y), PercentageMathRef.percentDiv(x, y));
     }
 
+    function testPercentAvg(
+        uint256 x,
+        uint256 y,
+        uint16 percentage
+    ) public {
+        vm.assume(percentage <= PERCENTAGE_FACTOR);
+
+        vm.assume(
+            percentage < PercentageMathRef.PERCENTAGE_FACTOR &&
+                x <= MAX_UINT256_MINUS_HALF_PERCENTAGE / (PercentageMathRef.PERCENTAGE_FACTOR - percentage)
+        );
+        vm.assume(percentage > 0 && y <= MAX_UINT256_MINUS_HALF_PERCENTAGE / percentage);
+
+        assertEq(
+            PercentageMath.percentAvg(x, y, percentage),
+            PercentageMathRef.percentMul(x, PercentageMathRef.PERCENTAGE_FACTOR - percentage) +
+                PercentageMathRef.percentMul(y, percentage)
```

Comment:
Why not using the reference implementation here ?

Reply:
because there is no reference implementation in aave's contracts

Reply:
Let's say that the contract `PercentageMathFunctionsRef` you just wrote is the ref, is it possible to reuse that ?